lib/iris/fileformats/pp_rules.py

```diff
@@ -968,30 +980,78 @@ def _all_other_rules(f):
 
     # Cross-sectional time values (--> vector coordinates)
     if (len(f.lbcode) == 5 and f.lbcode[-1] == 1 and f.lbcode.iy == 23):
+        epoch_hours_unit = f.time_unit('hours')
         dim_coords_and_dims.append(
             (DimCoord(
                 f.y,
                 standard_name='time',
-                units=Unit('days since 0000-01-01 00:00:00',
+                units=Unit(epoch_hours_unit,
                            calendar=iris.unit.CALENDAR_360_DAY),
                 bounds=f.y_bounds),
              0))
 
     if (len(f.lbcode) == 5 and f.lbcode[-1] == 1 and f.lbcode.ix == 23):
+        epoch_hours_unit = f.time_unit('hours')
         dim_coords_and_dims.append(
             (DimCoord(
                 f.x,
                 standard_name='time',
-                units=Unit('days since 0000-01-01 00:00:00',
+                units=Unit(epoch_hours_unit,
                            calendar=iris.unit.CALENDAR_360_DAY),
                 bounds=f.x_bounds),
              1))
 
+    # Check for cross-section over-specification.
+    if (len(f.lbcode) == 5 and
+            f.lbcode[-1] == 3 and
+            f.lbcode.ix == 23 and
+            hasattr(f, 'x')):
+        epoch_hours_unit = f.time_unit('hours')
+        t1_epoch_hours = epoch_hours_unit.date2num(f.t1)
+        if t1_epoch_hours != f.x[0]:
+            msg = ('PP cross-section file is over-specified for '
+                   'the time dimension, but the sets of values do not match.')
+            raise TranslationError(msg)
```

Comment:
> What if there are 50 fields in a file, and 49 of them are perfectly fine.

That doesn't excuse loading wrong data. If this happens the user should go back to the data supplier and request the file be fixed rather than Iris turning a blind eye. After all, no-one pays attention to warnings...